src/openrct2/world/sprite.c

```diff
@@ -767,9 +767,9 @@ void sprite_position_tween_all(float nudge)
             rct_xyz16 posB = _spritelocations2[i];
 
             sprite_set_coordinates(
-                posB.x + (sint16)((posA.x - posB.x) * nudge),
-                posB.y + (sint16)((posA.y - posB.y) * nudge),
-                posB.z + (sint16)((posA.z - posB.z) * nudge),
+                posB.x * alpha + posA.x * (1.0f - alpha),
+                posB.y * alpha + posA.y * (1.0f - alpha),
+                posB.z * alpha + posA.z * (1.0f - alpha),
```

Comment:
I think the compiler should be smart enough to figure that out. When I would precompute 1.0 - alpha I would need to pass another variable, it somewhat comes down to the same performance I believe.